### ISSUE TITLE
DOC/DX: add documentation for interactive examples and their maintenance

### ISSUE THREAD

**Issue by agriyakhetarpal** [numpy/numpy#27310]
### Issue with current documentation:

This is a short follow-up issue stemmed off from [gh-26745 (review)](https://github.com/numpy/numpy/pull/26745#pullrequestreview-2133346535) and [gh-26745 (comment)](https://github.com/numpy/numpy/pull/26745#issuecomment-2253036907) – once said PR is merged, the NumPy documentation for contributors and core developers should mention how to enable interactive examples with [`jupyterlite-sphinx`](https://jupyterlite-sphinx.readthedocs.io/en/stable/). This issue aims to track that process, with the information to be added pertaining to:
- interactive documentation and best practices on writing it, enabling it, and ensuring that it works as intended:
	- particularly, this means we should instruct docs contributors to write self-contained examples in the docstrings, as done in gh-26814
- `jupyterlite-sphinx`, links to its documentation, and instructions on how to enable/disable interactive examples on certain sections for the development version or released versions of the documentation via suitable modifications in [the `try_examples.json` file](https://jupyterlite-sphinx.readthedocs.io/en/latest/directives/try_examples.html#try-examples-json-configuration-file) for features or API sections that are not available with in-tree Pyodide wheels; and
- how updates in the documentation should be synced with updates from [Pyodide](https://github.com/pyodide/pyodide/) upstream, new minor releases for which usually carry package updates, usually updating core packages such as SciPy (https://github.com/scipy/scipy/issues/20918) and NumPy (https://github.com/pyodide/pyodide/pull/4925).
	- this is tied in with rebuilding the documentation for a particular version present at https://github.com/numpy/doc. Instructions for doing so are available in the repository's README.

### Additional context

- Please refer to this PR where updates to the SciPy documentation were made: https://github.com/scipy/scipy/pull/20843, and
- interactive documentation for SciPy was (previously) enabled in https://github.com/scipy/scipy/pull/20019

cc: @melissawm, @rgommers, @steppi

### Idea or request for content:

My idea is that the most appropriate place(s) to add documentation related to this should be in these pages and cross-linked with each other, based on the last point above:

1. https://github.com/numpy/numpy/blob/main/doc/RELEASE_WALKTHROUGH.rst, and
2. https://numpy.org/devdocs/dev/howto-docs.html#docstring-intro, and
3. perhaps a "Pyodide" section nearby https://numpy.org/devdocs/dev/howto_build_docs.html#instructions?

Additionally, some suggestions from @melissawm are as follows:

I would either link or get inspiration from [scipy.github.io/devdocs/dev/core-dev/index.html#wrapping-up](https://scipy.github.io/devdocs/dev/core-dev/index.html#wrapping-up) (for the release guide) and the "Interactive documentation" section on [scipy.github.io/devdocs/dev/contributor/rendering_documentation.html#rendering-documentation-locally-with-sphinx](https://scipy.github.io/devdocs/dev/contributor/rendering_documentation.html#rendering-documentation-locally-with-sphinx) 

<hr>

With some guidelines set on the interactive documentation goal and once gh-26745 gets merged, we could soon start looking at enabling interactivity in [the "User Guide" pages](https://numpy.org/devdocs/user/index.html#numpy-user-guide) and potentially for https://numpy.org/numpy-tutorials/ in the future. IMO, some of these points I have mentioned above could be better placed in part in a SPEC document rather than completely in the NumPy documentation, though.

**Comment by rootsmusic** [comment]
> some of these points I have mentioned above could be better placed in part in a SPEC document

@agriyakhetarpal ICYMI https://github.com/Quansight-Labs/czi-scientific-python-mgmt/issues/17
